lib/nolibc/string.c

```diff
@@ -829,8 +829,14 @@ char *strerror(int errnum)
 
 char *strcat(char *restrict dest, const char *restrict src)
 {
-	strcpy(dest + strlen(dest), src);
-	return dest;
+	char *tmp = dest;
+
+	while (*dest)
+		dest++;
+
+	while ((*dest++ = *src++) != '\0');
```

Comment:
nit: I think you can do 

```suggestion
	while (*dest++ = *src++');
```

Reply:
I would at least keep the additional parenthesis to signal that it's intentionally a assignment rather than comparison (There are compiler warnings for this)